### PR TITLE
Add dev fund addresses and set activation height for testnet NU4 activation.

### PR DESCRIPTION
**Pull request by ebfull**

### CODE REVIEW

**Comment by str4d** [comment]
(comment from @ebfull moved from top post)

Note that this PR can’t be merged until the Canopy activation height is set. This commit may need to be folded into the release PR instead.

File: src/chainparams.cpp [hunk 1]
```diff
@@ -353,7 +353,7 @@ class CTestNetParams : public CChainParams {
             uint256S("05688d8a0e9ff7c04f6f05e6d695dc5ab43b9c4803342d77ae360b2b27d2468e");
         consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nProtocolVersion = 170012;
         consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight =
-            Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT;
+            Consensus::NetworkUpgrade::1020200;
```
Comment (daira) [inline src/chainparams.cpp:356]:
```suggestion
       consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight = 1020500;
```

This is around 17:50Z on the 30th.

File: doc/release-notes.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ The code preparations for the Canopy network upgrade are finished and included i
 - [ZIP 214: Consensus rules for a Zcash Development Fund](https://zips.z.cash/zip-0214)
 - [ZIP 215: Explicitly Defining and Modifying Ed25519 Validation Rules](https://zips.z.cash/zip-0215)
 
-Canopy will activate on testnet at height TODO, and can also be activated at a specific height in regtest mode by setting the config option `-nuparams=0xe9ff75a6:HEIGHT`.
+Canopy will activate on testnet at height 1020200, and can also be activated at a specific height in regtest mode by setting the config option `-nuparams=0xe9ff75a6:HEIGHT`.
```
Comment (therealyingtong) [inline doc/release-notes.md:18]:
```suggestion
Canopy will activate on testnet at height 1020500, and can also be activated at a specific height in regtest mode by setting the config option `-nuparams=0xe9ff75a6:HEIGHT`.
```

File: src/chainparams.cpp [hunk 3]
```diff
@@ -352,8 +352,7 @@ class CTestNetParams : public CChainParams {
         consensus.vUpgrades[Consensus::UPGRADE_HEARTWOOD].hashActivationBlock =
             uint256S("05688d8a0e9ff7c04f6f05e6d695dc5ab43b9c4803342d77ae360b2b27d2468e");
         consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nProtocolVersion = 170012;
-        consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight =
-            Consensus::NetworkUpgrade::1020500;
+        consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight = 1029140;
```
Comment (str4d) [inline src/chainparams.cpp:355]: One week of blocks is `7 * 24 * 48 = 8064`, while the difference in this commit is `1029140 - 1020500 = 8640`. Typo?

In any case, let's round the activation to the nearest 500.

Comment (therealyingtong) [reply src/chainparams.cpp]: updated activation height to `1028500`, which gives us a difference of `1028500 - 1020500 = 8000`.

File: src/chainparams.cpp [hunk 4]
```diff
@@ -352,8 +352,7 @@ class CTestNetParams : public CChainParams {
         consensus.vUpgrades[Consensus::UPGRADE_HEARTWOOD].hashActivationBlock =
             uint256S("05688d8a0e9ff7c04f6f05e6d695dc5ab43b9c4803342d77ae360b2b27d2468e");
         consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nProtocolVersion = 170012;
-        consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight =
-            Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT;
+        consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight = 1028500;
```
Comment (daira) [inline src/chainparams.cpp:355]: This is Wednesday August 5th at 15:47 UTC. Seems fine to me, although it's a little early on the west coast.

**Comment by daira** [comment]
The activation height is a consensus change, so this needs one more ACK.

**Comment by daira** [comment]
@zkbot r+

**Comment by daira** [comment]
Sorry, didn't intend to rerequest a review from @nuttycom, only from @therealyingtong and @str4d.

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**